Info-Orbs/src/model/webDataElementArcModel.cpp

```diff
@@ -0,0 +1,134 @@
+#include "model/webDataElementArcModel.h"
+
+int32_t WebDataElementArcModel::getX() {
+    return m_x;
+}
+
+void WebDataElementArcModel::setX(int32_t x) {
+    if (m_x != x) {
+        m_x = x;
+        m_changed = true;
+    }
+}
+
+int32_t WebDataElementArcModel::getY() {
+    return m_y;
+}
+
+void WebDataElementArcModel::setY(int32_t y) {
+    if (m_y != y) {
+        m_y = y;
+        m_changed = true;
+    }
+}
+
+int32_t WebDataElementArcModel::getRadius() {
+    return m_radius;
+}
+
+void WebDataElementArcModel::setRadius(int32_t radius) {
+    if (m_radius != radius) {
+        m_radius = radius;
+        m_changed = true;
+    }
+}
+
+int32_t WebDataElementArcModel::getInnerRadius() {
+    return m_innerRadius;
+}
+
+void WebDataElementArcModel::setInnerRadius(int32_t radius) {
+    if (m_innerRadius != radius) {
+        m_innerRadius = radius;
+        m_changed = true;
+    }
+}
+
+uint32_t WebDataElementArcModel::getAngleStart() {
+    return m_angleStart;
+}
+
+void WebDataElementArcModel::setAngleStart(uint32_t angle) {
+    if (m_angleStart != angle) {
+        m_angleStart = angle;
+        m_changed = true;
+    }
+}
+
+uint32_t WebDataElementArcModel::getAngleEnd() {
+    return m_angleEnd;
+}
+
+void WebDataElementArcModel::setAngleEnd(uint32_t angle) {
+    if (m_angleEnd != angle) {
+        m_angleEnd = angle;
+        m_changed = true;
+    }
+}
+
+int32_t WebDataElementArcModel::getBackgroundColor() {
+    return m_background;
+}
+
+void WebDataElementArcModel::setBackgroundColor(int32_t background) {
+    if (m_background != background) {
+        m_background = background;
+        m_changed = true;
+    }
+}
+
+void WebDataElementArcModel::setBackgroundColor(String background) {
+    setBackgroundColor(Utils::stringToColor(background));
+}
+
+void WebDataElementArcModel::setColor(int32_t color) {
+    if (m_color != color) {
+        m_color = color;
+        m_changed = true;
+    }
+}
+
+void WebDataElementArcModel::setColor(String color) {
+    setColor(Utils::stringToColor(color));
+}
+
+int32_t WebDataElementArcModel::getColor() {
+    return m_color;
+}
+
+void WebDataElementArcModel::parseData(const JsonObject& doc, int32_t defaultColor, int32_t defaultBackground) {
+    // Serial.println("Parsing Data");
+
+    if (doc.containsKey("x")) {
```

Comment:
I agreee. What do they recommend replacing it with?

Reply:
https://arduinojson.org/v7/api/jsondocument/containskey/
For int and the like, you can use `is<int>()`,
for strings you can just read it and it will be a `nullptr` if there is no value:
```cpp
const char* city = doc["city"];
if (city)
  Serial.println(city);
```

For example, reading the color previously:
```cpp
if (doc[i].containsKey("color")) {
    data->setDataColor(doc[i]["color"].as<String>());
```
Could be written as:
```cpp
if (const char* color = doc[i]["color"]) {
    data->setDataColor(color);
```

Reply:
Using is<int>() is fine. But creating a variable that is only used for a function call seems like a bad idea. But we are using arduinojson v7 already. So I suppose it needs to be done.

Reply:
I have started making this change but strings will need more testing. So it will be a followup PR

Reply:
> Using is() is fine. But creating a variable that is only used for a function call seems like a bad idea. But we are using arduinojson v7 already. So I suppose it needs to be done.


> But creating a variable that is only used for a function call seems like a bad idea. 

It is not, in this case it only has to do the lookup once instead of twice so overall it will be _way_ cheaper.